plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteCaseInsensitiveMapping.java

```diff
@@ -0,0 +1,248 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.trino.plugin.jdbc.BaseCaseInsensitiveMappingTest;
+import io.trino.testing.MaterializedRow;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.SqlExecutor;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.trino.plugin.ignite.IgniteQueryRunner.createIgniteQueryRunner;
+import static io.trino.plugin.jdbc.mapping.RuleBasedIdentifierMappingUtils.createRuleBasedIdentifierMappingFile;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+// With case-insensitive-name-matching enabled colliding schema/table names are considered as errors.
+// Some tests here create colliding names which can cause any other concurrent test to fail.
+@Test(singleThreaded = true)
+public class TestIgniteCaseInsensitiveMapping
+        extends BaseCaseInsensitiveMappingTest
+{
+    private Path mappingFile;
+    private TestingIgniteServer igniteServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        mappingFile = createRuleBasedIdentifierMappingFile();
+        igniteServer = TestingIgniteServer.getInstance();
+        return createIgniteQueryRunner(
+                igniteServer,
+                ImmutableMap.of(),
+                ImmutableMap.<String, String>builder()
+                        .put("case-insensitive-name-matching", "true")
+                        .put("case-insensitive-name-matching.config-file", mappingFile.toFile().getAbsolutePath())
+                        .put("case-insensitive-name-matching.config-file.refresh-period", "1ms") // ~always refresh
+                        .buildOrThrow(),
+                ImmutableList.of());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy()
+    {
+        TestingIgniteServer.shutDown();
+    }
+
+    @Override
+    protected AutoCloseable withSchema(String schemaName)
+    {
+        return null;
+    }
+
+    @Override
+    protected String quoted(String name)
+    {
+        String identifierQuote = "`";
+        name = name.replace(identifierQuote, identifierQuote + identifierQuote);
+        return identifierQuote + name + identifierQuote;
+    }
+
+    @Override
```

Comment:
why do we need to override each (most) of tests here?

Reply:
It's because of the constraints:
- Currently we do not support creating schema for Ignite connector, and Ignite cluster not support this operation in sql way neither. 
- Ignite only has ``public`` schema by default.
- For the Ignite table, it's need at least one primary key and one non-primary.